### ISSUE TITLE
SearchBox not handling wildcard char (*) as expected

### ISSUE THREAD

**Issue by X1M3N** [microsoft-search/pnp-modern-search#2223]
Hello there,

We have a search page using refiners, search box, and search results.
All 3 work correctly together.

However, using the wildcard search character (*) in the search box is not giving the expected result.

We have 2 documents with a metadata called "Numéro":
![image](https://user-images.githubusercontent.com/72074119/178460186-9977fe6f-dfc9-44eb-9cd5-8ca95f777ac3.png)

When we use the global site search (not the pnp web part), and type the following search, documents are retrieved correctly:
![image](https://user-images.githubusercontent.com/72074119/178460530-a7b65bdc-f306-4d2c-98f9-afb969e5c7d6.png)

If we type the same search in the search box, no results are shown:
![image](https://user-images.githubusercontent.com/72074119/178460728-02cba662-5bb3-4fc0-940d-220742c15126.png)

Adding a '4' in before the wildcard works:
![image](https://user-images.githubusercontent.com/72074119/178461025-0904b610-d962-4baf-86bb-ae11d010cef5.png)

Is there anything we seem to be missing?

Thanks!

**Comment by vherrlein** [comment]
Same kind of trouble if the SearchBox is configured to open a new page url containing an hashtag.

Page Url: https://myurl/my-spa/#/route/path
QueryString Param: v
Expected Url: https://myurl/my-spa/#/route/path?v=valueSearchBox

The issue seems coming from the UrlHelper: https://github.com/microsoft-search/pnp-modern-search/blob/af71d24d76fdda4dcd6f55b8b0197e656eb580c8/search-parts/src/helpers/UrlHelper.ts#L70

**Comment by X1M3N** [comment]
@vherrlein thank you for your input.
However, I am not sure I understand if I should do something on my end or not...

Thanks again.

**Comment by FranckyC** [comment]
Hi @X1M3N, what data source are you using? SharePoint or MS Search? also, can you share your settings regarding search box connection and query template value? Thanks!

**Comment by Kyou04** [comment]
Hello @FranckyC , 
I'm a collegue of X1M3N . 

We use Sharepoint Search 
Here is the setting. 

![image](https://user-images.githubusercontent.com/54059745/182591465-c74404a9-8717-4bda-9afb-af469d795563.png)

Best regards.

**Comment by wobba** [comment]
You query for the term `879.654`, where the `.` will be tokenized away, so not sure that query will work at all.

**Comment by X1M3N** [comment]
> You query for the term `879.654`, where the `.` will be tokenized away, so not sure that query will work at all with a wildcard at the end. So an KQL API issue, not web part related the way I see it.

Thank you for your response.

Can you explain why the pnp search is not returning results when the global search is returning results (see my first post)?
Is it related to the way the SharePoint search API responds? If so, could there be any workaround?

Thanks again.

**Comment by wobba** [comment]
Should be the same... API calls are a bit different and could be the hyphen in the mp name you have. Can try add an alias without a hyphen and try? You should also try via the Search Query Tool or the classic search center.

**Comment by X1M3N** [comment]
> Should be the same... API calls are a bit different and could be the hyphen in the mp name you have. Can try add an alias without a hyphen and try? You should also try via the Search Query Tool or the classic search center.

I did try through the standard SharePoint search, it is giving the right results (see one of my screenshots above).

You mention the mp name, but it does not contain a hyphen.
Here is the configuration:
![image](https://user-images.githubusercontent.com/72074119/194248675-bd5c1793-843a-42c6-a624-3b6181e2d160.png)

Also, I tried with the RefinableString05 name directly, which is giving the right results:
![image](https://user-images.githubusercontent.com/72074119/194248867-dd73f750-29cb-41c8-8f63-28476ec1b36a.png)

Are we doing something wrong?
